cfpb/cfgov-refresh
# Upgrade cf-typography to 4.8.0 to remove Avenir Next Italic web font

Upgrades cf-typography to remove Avenir Next Italic web font. Removing this web font improves page performance for affected pages. The browser italicizes the Avenir Next Regular web font instead of loading an additional font file. 

Pages with this reduction in page weight have shown an average decrease of 1.07 seconds in the first meaningful paint time, for desktop users on a broadband connection. For a simulated mobile connection, the average decrease for this metric was 3.19s. See more notes on the performance impact of this change below.

## Changes

- Upgrade to latest cf-typography version
- The main affected pages are Ask CFPB answer pages and Owning a Home (both use the Feedback Form module, which has italic text).

## Testing

1. check out branch
1. edit `main.less` line 30 to: `@use-font-cdn: true;` to view web fonts locally
1. `./frontend.sh`
1. `./runserver.sh`
1. Test that the following key pages do not load Avenir Next Italic web font (check Dev Tools > Network Tab > click "Font" filter and hit reload):  
  - http://localhost:8000/owning-a-home
  - http://localhost:8000/ask-cfpb/what-should-i-do-when-a-debt-collector-contacts-me-en-1695/

## Screenshots
![image](https://user-images.githubusercontent.com/702526/34167148-e7b41b1c-e4ae-11e7-82e4-d41091e2f0bd.png)

Additional screenshots and comparisons of examples from cfgov-refresh are documented at https://GHE/CFGOV/platform/issues/2347#issuecomment-140554

## Notes on performance

- Pages that use the italic font currently will see a 19kb reduction in page weight once this is deployed.
- Pages with this reduction in page weight have shown an average decrease of 1.07 seconds in the first meaningful paint time, for desktop users on a broadband connection. For a simulated mobile connection, the average decrease for this metric was 3.19s. [See page load metrics on GHE](https://GHE/CFGOV/platform/issues/2347#issuecomment-140861).
- Few pages use the italic font style, so most pages will not see a performance gain. Key pages that do use the italic style include Owning a Home and Ask CFPB (on the Feedback form module) and the [Learn more](www.consumerfinance.gov/learnmore/) page, all of which receive significant traffic.

## Checklist
